docs/graph_engine/temporal.rst

```diff
@@ -0,0 +1,144 @@
+.. code-block:: json
+
+	{
+		"node_id": 42,
+		"node_type": 0,
+		"node_weight": 1.0,
+		"created_at": 13,
+		"removed_at": 23,
+		"edge": [
+			{
+				"src_id": 42,
+				"dst_id": 99,
+				"edge_type": 0,
+				"weight": 2.0,
+				"created_at": 13,
+				"removed_at": 15
+			},
+			{
+				"src_id": 42,
+				"dst_id": 99,
+				"edge_type": 0,
+				"weight": 2.0,
+				"created_at": 17
+			},
+			{
+				"src_id": 42,
+				"dst_id": 101,
+				"edge_type": 1,
+				"weight": 1.0,
+				"removed_at": 19
+			}
+		]
+	}
+
+The node with ID 42 has two outgoing edges with types 0 and 1. The first edge exists during intervals [13, 15) and [17, 23), while the second edge is created with the node at time 13 and removed at 19. If the created_at field is not specified, it defaults to the created_at of the node. Similarly, the default value for removed_at is the same as the removed_at of the node.
+
+Temporal Features
+-----------------
+
+Temporal features are node or edge features that can change values over time. To support this behavior, the schema for existing features is expanded. For binary and dense features, the format includes a union of two types:
+
+A list of values for features that are present during the node/edge lifetime.
+A list of objects with three fields: values, created_at, and removed_at.
+The default values for created_at and removed_at are similar to those for nodes and edges. Sparse features are extended in a similar way, with the accepted types being a union of a sparse feature object and a list of such objects. The following example demonstrates the use of temporal features:
+
+.. code-block:: json
+
+	{
+		"node_id": 42,
+		"node_type": 0,
+		"node_weight": 1.0,
+		"binary_feature": {
+			"0": [
+				{"values": ["microsoft"], "created_at": 13, "removed_at": 20},
+				{"values": ["xbox"], "created_at": 20}
+			]
+		},
+		"created_at": 13,
+		"removed_at": 23,
+		"edge": [
+			{
+				"src_id": 42,
+				"dst_id": 99,
+				"edge_type": 0,
+				"weight": 2.0,
+				"created_at": 13,
+				"removed_at": 23,
+				"sparse_int32_feature": {
+					"0": [
+						{"coordinates": [[5, 13], [7, 25]], "values": [-1, 1024], "removed_at": 21},
+						{"coordinates": [[4, 2], [1, 3]], "values": [2, 4], "created_at": 21}
+					],
+					"1": {"coordinates": [13, 42], "values": [1, 1]}
+				}
+			}
+		]
+	}
+
+In the example above, there are three temporal features: a binary feature belonging to the node with ID 42 and two sparse integer features attached to an edge with source node 42 and destination node 99. Feature with ID 0 has two temporal values assigned to it, split around time 21. Feature with ID 1 is present during the entire life of the edge, from 13 to 23.
+
+Edge List Format
+================
+
+The Edge List format supports temporal edges and nodes, but not features. The original order of the node first and then all outgoing edges remains the same. EdgeInfo and NodeInfo are extended with new optional items to include creation and removal times:
```

Comment:
It is possible to make edge list format support temporal features, you should do this.

Reply:
Not in this PRs